core/src/klio_core/options.py

```diff
@@ -127,25 +139,127 @@ def wrapper(func):
     return wrapper
 
 
-def override(func):
+#####
+# common options
+#####
+def direct_runner(func):
     return click.option(
-        "-O",
-        "--override",
-        default=[],
-        multiple=True,
-        help="Override a config value, in the form ``key=value``.",
+        "--direct-runner",
+        default=False,
+        is_flag=True,
+        help="Run the job locally via the DirectRunner.",
     )(func)
 
 
-def template(func):
+def update(func):
     return click.option(
-        "-T",
-        "--template",
-        default=[],
-        multiple=True,
+        "--update/--no-update",
+        default=None,
+        is_flag=True,
+        help="[Experimental] Update an existing streaming Cloud Dataflow job.",
+    )(func)
+
+
+def show_logs(func):
+    return click.option(
+        "--show-logs",
+        default=False,
+        show_default=True,
+        is_flag=True,
+        help="Show a job's logs while profiling.",
+    )(func)
+
+
+#####
+# options for `klioexec profile memory
+#####
+def interval(func):
+    return click.option(
+        "--interval",
+        default=0.1,
+        show_default=True,
+        type=float,
+        help="Sampling period (in seconds).",
+    )(func)
+
+
+def include_children(func):
+    return click.option(
+        "--include-children",
+        default=False,
+        show_default=True,
+        is_flag=True,
+        help="Monitor forked processes as well (sums up all process memory).",
+    )(func)
+
+
+def multiprocess(func):
+    return click.option(
+        "--multiprocess",
+        default=False,
+        show_default=True,
+        is_flag=True,
         help=(
-            "Set the value of a config template parameter"
-            ", in the form ``key=value``.  Any instance of ``${key}`` "
-            "in ``klio-job.yaml`` will be replaced with ``value``."
+            "Monitor forked processes creating individual plots for each "
+            "child."
+        ),
+    )(func)
+
+
+def plot_graph(func):
+    return click.option(
+        "-g",
+        "--plot-graph",
+        default=False,
+        show_default=True,
+        is_flag=True,
+        help=(
+            "Plot memory profile using matplotlib. Saves to "
+            "klio_profile_memory_<YYYYMMDDhhmmss>.png."
+        ),
+    )(func)
+
+
+#####
+# options for `klio job profile memory-per-line`
+#####
+def maximum(func):
+    return click.option(
+        "--maximum",
+        "get_maximum",
+        default=False,
+        show_default=True,
+        is_flag=True,
+        cls=MutuallyExclusiveOption,
+        mutually_exclusive=["per_element"],
+        help=(
+            "Print maximum memory usage per line in aggregate of all input "
+            "elements process."
         ),
     )(func)
+
+
+def per_element(func):
+    return click.option(
+        "--per-element",
+        default=False,
+        show_default=True,
```

Comment:
In the cli version of this option, it's `show_default=True` (as it is here); in the [exec version of this option](https://github.com/spotify/klio/blob/master/exec/src/klio_exec/options.py#L226), it's `show_default="True"` (`bool` vs `str`). They are two different behaviors: the first will show that the default is set to `False` as declared right above it; the second will show the default is "True" because it's a string. Also notice the [comment](https://github.com/spotify/klio/blob/master/exec/src/klio_exec/options.py#L226) for the exec option - it tells me that the logic for handling the option will actually be a little different than how the CLI will handle it. 

We should keep the original behavior, however it would be good to not have two `per_element` options that are otherwise quite similar. This can be achieved by defining an inner function, similar to how it's done with [`job_name`](https://github.com/spotify/klio/blob/master/cli/src/klio_cli/options.py#L197-L205):

```py
# in core/src/klio_core/options.py
def per_element(*args, **kwargs):
    def wrapper(func):
        return click.option( 
            "--per-element",
            default=False,
            is_flag=True,
            cls=MutuallyExclusiveOption,
            mutually_exclusive=["maximum"],
            help="Print memory usage per line for each input element processed.",
            **kwargs,  # added (and no more show_default keyword arg)
        )(func)

    if args:
        return wrapper(args[0])
    return wrapper

# in exec/src/klio_exec/cli.py
# for all the commands that use the per_element option:
@core_options.per_element(show_default="True")
...

# in cli/src/klio_cli/cli.py
# for all the commands that use the per_element option:
@core_options.per_element(show_default=True)
...
```